src/main/java/com/okta/sdk/framework/ApiClient.java

```diff
@@ -45,18 +51,19 @@
     private AtomicReference<HttpResponse> lastHttpResponseRef = new AtomicReference<HttpResponse>();
 
     ////////////////////////////////////////////
-    // CONSTRUCTORS
+    // CONSTRUCTOR
     ////////////////////////////////////////////
 
     public ApiClient(ApiClientConfiguration config) {
-        this(HttpClientBuilder.create()
-                        .setUserAgent("OktaSDKJava_v" + Utils.getSdkVersion())
-                        .disableCookieManagement().build(),
-                config);
-    }
+        PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+        connectionManager.closeExpiredConnections();
+        connectionManager.closeIdleConnections(TTL, TimeUnit.SECONDS);
```

Comment:
I don't think calling `connectionManager.closeExpiredConnections()` or `connectionManager.closeIdleConnections(TTL, TimeUnit.SECONDS)` will do anything useful at this point. According to the [PoolingHttpClientConnectionManager Javadoc](http://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/conn/PoolingHttpClientConnectionManager.html#closeIdleConnections(long,%20java.util.concurrent.TimeUnit)), these methods close idle or expired connections at the time they are called. But right after creating the `PoolingHttpClientConnectionManager` there are not even any connections yet.

Normally these methods would be called periodically from a background thread (see section 8 of this [HttpClient guide](http://www.baeldung.com/httpclient-connection-management).

Also I don't think it's necessary to call both: the documentation for `closeIdleConnections` states "All expired connections will also be closed."

Anyway the easiest way to set a persistent connection TTL appears to be [`HttpClientBuilder.setConnectionTimeToLive`](http://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/client/HttpClientBuilder.html#setConnectionTimeToLive(long,%20java.util.concurrent.TimeUnit)).